src/monitor/node_active_protocol.c

```diff
@@ -170,6 +170,14 @@ register_node(PG_FUNCTION_ARGS)
 	LockNodeGroup(formationId, currentNodeState.groupId, ExclusiveLock);
 
 	pgAutoFailoverNode = GetAutoFailoverNode(nodeName, nodePort);
+	if (pgAutoFailoverNode == NULL) {
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("node %s:%d with dbname \"%s\" could not be registered in "
+						"formation \"%s\", could not get matching node",
```

Comment:
Let's improve the error message. The node we're failing to find now is the one we just did an INSERT for in the previous lines of the code...